R/utils.R

```diff
@@ -157,17 +157,31 @@ with_rprofile_user <- function(new, code) {
 
 untar <- function(tarfile, ...) {
   if (os_type() == "windows") {
-    status <- try(utils::untar(tarfile, extras = "--force-local", ...))
-    if(inherits(status, "try-error") || status != 0){
+    tarhelp <- system2("tar", "--help", stdout = TRUE, stderr = TRUE)
+    forcelocal <- any(grepl("--force-local", tarhelp))
+    if (forcelocal)  {
+      status <- try(
+        suppressWarnings(utils::untar(tarfile, extras = "--force-local", ...)),
+        silent = TRUE)
+      if (inherits(status, "try-error") ||
+          is_error_status(status) || is_error_status(attr(status, "status"))) {
+        message("External tar failed with `--force-local`, trying without")
         utils::untar(tarfile, ...)
```

Comment:
There are three cases that all end with `utils::untar(tarfile, ...)`. So we should be able to rewrite the case that returns the status with an early return and have the rest of the cases fall through.

Reply:
I tried that but it was actually less readable than the current code. You cannot mark "not done  yet"  with `status =  NULL`, because maybe that's  a legit return value. Do you need a `done` variable, plus `status` ... anyway, I'll give it another try.